### ISSUE TITLE
1.0.1 - CPU usage 250% but nothing happens

### ISSUE THREAD

**Issue by tessus** [roman-r-m/joplin-plugin-table-formatter#2]
When I click `Edit > Format Table`, I get thousands of messages in the console:

```
00:40:40: CommandService::execute: formatTable
main-html.js:48 00:40:40: PluginRunner: Got message (3): joplin.commands.execute formatTable
main-html.js:48 00:40:40: CommandService::execute: formatTable
main-html.js:48 00:40:40: PluginRunner: Got message (3): joplin.commands.execute formatTable
main-html.js:48 00:40:40: CommandService::execute: formatTable
main-html.js:48 00:40:40: PluginRunner: Got message (3): joplin.commands.execute formatTable
main-html.js:48 00:40:40: CommandService::execute: formatTable
main-html.js:48 00:40:40: PluginRunner: Got message (3): joplin.commands.execute formatTable
main-html.js:48 00:40:40: CommandService::execute: formatTable
main-html.js:48 00:40:40: PluginRunner: Got message (3): joplin.commands.execute formatTable
```

When I close Joplin, I get the following:

```
A JavaScript error occurred in the main process

Uncaught Exception:
TypeError: Object has been destroyed
    at IpcMainImpl.<anonymous> (/Users/tessus/data/work/joplin/packages/app-desktop/ElectronAppWrapper.js:178:21)
    at IpcMainImpl.emit (events.js:310:20)
    at WebContents.<anonymous> (electron/js2c/browser_init.js:157:8362)
    at WebContents.emit (events.js:310:20)
```

There are no errors in the `log.txt`.

```
Joplin 1.8.0 (dev, darwin)

Client ID: e354a932e6e143c782264aad4112b674
Sync Version: 2
Profile Version: 34
Keychain Supported: Yes

Revision: 131a7e736 (dev)
```

**Comment by roman-r-m** [comment]
Thanks for raising this. Does it occur with any particular note text?

**Comment by tessus** [comment]
Argh, I knew I forgot something....  Sorry about that.

```
|head1|heassadad2|head3|
|-----|-----|-----|
|colfdfdafaf1 |col2 |col3 |
|col1 |col2 |col3 |  
```

But also happens with this

```
head1 | head2 | head3
-|-|-
col1 |col2 |col3
col1 |col2 |col3
```

Will be heading to bed now. It's way past my bedtime.

**Comment by roman-r-m** [comment]
~~I can reproduce it, will take a look later today.~~

Actually no, I can't. I reproduced it when I had some uncommitted changes - was trying to backport it to the current released version by looking at https://github.com/laurent22/joplin/pull/4406/files

Any chance you had local changes?

**Comment by roman-r-m** [comment]
I also see slightly different log messages:
```
12:45:26: PluginRunner: Got message (3): joplin.commands.execute editor.execCommand {name: "formatTable"}
main-html.js:48 12:45:26: CommandService::execute: editor.execCommand {name: "formatTable"}
main-html.js:48 12:45:26: CodeMirror: execCommand {name: "editor.execCommand", value: {…}}
```

Might be something specific to your fork. I'll clone it and check.

**Comment by tessus** [comment]
> Any chance you had local changes?

No, this is the dev branch.

**Comment by tessus** [comment]
> Might be something specific to your fork.

I don't understand. I'm using upstream/dev.

P.S.: Oh, forget my fork. I use my fork exclusively for PRs, which I would not even have to, since I can create branches in the source repo. My fork does not even have the dev branch.

**Comment by roman-r-m** [comment]
Looks like it was I who had local changes when publishing the plugin, this was causing the infinite loop. This is now fixed in 1.1.0.

However, I have not yet implemented support for tables without the leading and trailng pipes like in your 2nd example. But at least your CPU is now safe.

**Comment by tessus** [comment]
Awesome. This fixed it. I found another issue, but will open a separate ticket. Many devs want separate tickets....
